java/runtime/src/main/java/org/ray/runtime/object/NativeRayObject.java

```diff
@@ -9,8 +11,15 @@
   public byte[] metadata;
 
   public NativeRayObject(byte[] data, byte[] metadata) {
+    Preconditions.checkNotNull(data);
+    Preconditions.checkNotNull(metadata);
```

Comment:
should `metadata` be a zero-length array when there's no metadata? If so, we can document this.